### PR TITLE
Fixed PGVector issue when SQLAlchemy is the driver

### PR DESCRIPTION
**Pull request by ranaroussi**
Using SQLAlchemy as the driver for PGVector's `connection_string_from_db_params` function fails with the following error:

```
NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:postgresql.sqlalchemy
```

This PR removes the `+{driver}` when `sqlalchemy` is used as the driver.

Additionally, this PR also adds support for query params in the connection string via the optional `query_params` param. 

Example:

```python
CONNECTION_STRING = PGVector.connection_string_from_db_params(
    driver="sqlalchemy", 
    host="dbhost.com",
    port=5432,
    database="dbname",
    user="user",
    password="pass",
    query_params={ # ← this is new
        "sslmode": "require"
    }
)'

print(CONNECTION_STRING)

# 'postgresql://user:pass@dbhost.com:5432/dbname?sslmode=require', 
```

Tested on Postgres 14, 15, and Supabase

@dev2049 ☝️

### CODE REVIEW

File: langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -593,6 +594,9 @@ def connection_string_from_db_params(
         database: str,
         user: str,
         password: str,
+        **kwargs
```
Comment (baskaryan) [inline langchain/vectorstores/pgvector.py:597]: should we just call this query_params and make it an optional dict if that's what it's used for

Comment (ranaroussi) [reply langchain/vectorstores/pgvector.py:597]: That's a great idea. I'll push an updated PR

**Comment by leo-gan** [comment]
@ranaroussi Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**